sklearn/ensemble/tests/test_bagging.py

```diff
@@ -586,19 +606,268 @@ def predict(self, X):
+
+
+@pytest.mark.parametrize("estimator_cls", [BaggingClassifier, BaggingRegressor])
+def test_sample_weight_none_and_base_estimator_supports_it_indirectly(estimator_cls):
+    # Even though "sample_weight" is not in the fit signature, it is still
+    # passed. This test exists because bagging used to determine if
+    # sample_weight should be passed by inspecting the fit signature.
+    from sklearn.utils.metadata_routing import RequestType
+
+    class MyEstimator(BaseEstimator):
+        # This attribute indicates that the estimator indeed supports
+        # sample_weight
+        __metadata_request__fit = {"sample_weight": RequestType.ERROR_IF_PASSED}
+
+        def fit(self, X, y, **kwargs):
+            assert kwargs.get("sample_weight") is not None
+            return self
+
+        def predict(self, X):
+            return np.ones(len(X))
+
+    # Using iris even for regression, which is fine for the purpose of this test
+    X, y = iris.data, iris.target
+    # sample bagging is achieved through the use of sample weights because the
+    # base estimator supports it
+    base_estimator = MyEstimator()
+    estimator = estimator_cls(base_estimator)
+
+    # there are no warnings and no errors
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        estimator.fit(X, y)
+
+
+@pytest.mark.parametrize("estimator_cls", [BaggingClassifier, BaggingRegressor])
+def test_sample_weight_none_and_base_estimator_does_not_support_it(estimator_cls):
+    # sample bagging is achieved through indexing because the base estimator
+    # does not support it
+
+    # Using iris even for regression, which is fine for the purpose of this test
+    X, y = iris.data, iris.target
+    n_samples = len(y)
+
+    class MyEstimator(BaseEstimator):
+        def fit(self, X, y, **fit_params):
+            # check that the indexing method was used to achieve subsampling,
+            # i.e. no sample_weight are passed but instead fewer samples
+            assert "sample_weight" not in fit_params
+            assert len(y) == 0.5 * n_samples
+            return self
+
+        def predict(self, X):
+            return np.ones(len(X))
+
+    base_estimator = MyEstimator()
+    estimator = estimator_cls(base_estimator, max_samples=0.5)
+
+    # there are no warnings and no errors
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        estimator.fit(X, y)
+
+
+@pytest.mark.parametrize("estimator_cls", [BaggingClassifier, BaggingRegressor])
+def test_sample_weight_not_none_and_base_estimator_supports_it_directly(estimator_cls):
+    # sample bagging is achieved through the use of sample weights because the
+    # base estimator supports it (but has to request it)
+
+    # Using iris even for regression, which is fine for the purpose of this test
+    X, y = iris.data, iris.target
+    sample_weight = np.ones_like(y)
+
+    base_estimator = _weighted(CheckingClassifier(expected_sample_weight=False))
```

Comment:
Should this be `expected_sample_weight=True`? The comment above says "is achieved through the use of sample weights"

Reply:
Good catch.